coil-compose-base/src/main/java/coil/compose/AsyncImagePainter.kt

```diff
@@ -282,9 +288,9 @@ class AsyncImagePainter internal constructor(
     }
 
     @OptIn(ExperimentalCoroutinesApi::class)
-    override fun onRemembered() {
+    override fun onRemembered() = trace("AsyncImagePainter.onRemembered") {
```

Comment:
I'm thinking whether the usage of `onRemembered` is used actually? This is manually called immediatelly in composition and then compose framework calls it again after the composition and it always just returns with the check on `rememberScope != null`. 

I don't fully understand (yet) the child painter and it's relation to `RememberObserver`, but feels like it could call them differently as well?

Reply:
I'm not too sure of Compose's behaviour, but I added this in on `onRemembered` since we want to stop any ongoing requests in `onForgotten`. This keeps the symmetry in case the painter is re-remembered after being forgotten.

The child painter is usually either a `CrossfadePainter` if crossfading is enabled or a `DrawablePainter`, which wraps the `Drawable` returned by the `ImageLoader`.